tests/test_recreation.py

```diff
@@ -140,90 +141,157 @@ def test_basic_operation(self):
             file_manager.read(1234)
 
 
-class TestRecServer(unittest.TestCase):
-    """Tests that set up local rec server on a port and call through."""
+class TestRecServerLoop(unittest.TestCase):
+    """Tests that use the rec server execute loop running in another process."""
 
     def setUp(self):
         """Setup workspace."""
+        from natcap.invest.recreation import recmodel_server
+
         self.workspace_dir = tempfile.mkdtemp()
         self.resampled_data_path = os.path.join(
             self.workspace_dir, 'resampled_data.csv')
         _resample_csv(
             os.path.join(SAMPLE_DATA, 'sample_data.csv'),
             self.resampled_data_path, resample_factor=10)
 
+        # attempt to get an open port; could result in race condition but
+        # will be okay for a test. if this test ever fails because of port
+        # in use, that's probably why
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.bind(('', 0))
+        self.port = sock.getsockname()[1]
+        sock.close()
+        sock = None
+
+        server_args = {
+            'hostname': 'localhost',
+            'port': self.port,
+            'raw_csv_point_data_path': self.resampled_data_path,
+            'cache_workspace': self.workspace_dir,
+            'min_year': 2008,
+            'max_year': 2015,
+            'max_points_per_node': 200,
+        }
+
+        self.server_process = multiprocessing.Process(
+            target=recmodel_server.execute, args=(server_args,), daemon=False)
+        self.server_process.start()
+        time.sleep(5)  # need a few seconds for the server to be ready

```

Comment:
I added a comment with these options, in case it does become flaky and we have to revisit it. I'd prefer to keep it simple with a `time.sleep` if possible, though